### ISSUE TITLE
The original code reproduces performance issues

### ISSUE THREAD

**Issue by HE-SHUANG-LI** [ricardoborsoi/Unmixing_with_Deep_Generative_Models#1]
# Asking for help
when I try to run the example1.m file, the code performance always not so good as the performance of the paper in windows or in ubuntu.  So I hope to get help from the original author in this issue.  The details are as folloing. 

Looking forward to your prompt reply.

the python result：
![UMNTNPD%MJ%S($Z}NWND_~K](https://user-images.githubusercontent.com/53934251/163720444-61758a79-334e-4e2c-ad29-06dfb08b7596.png)

## MY environment: 
    1. (anaconda + tensorflow2.2.0 + keras2.4.3) python3.6.13：train vaes model（vae_keras_ems_train.py） 
    2. MATLAB2020b 
    3. No GPU and just cpu

## output performance:

![1](https://user-images.githubusercontent.com/53934251/163720302-de536c8c-d9e3-4672-af5c-ab4b49a78bbf.png)

RMSE for Abundances
FCLS.....: 0.2854 
PLMM.....: 0.2604 
ELMM.....: 0.2554 
GLMM.....: 0.2480 
DeepGen..: 0.2677 

RMSE for Endmembers
PLMM.....: 0.1075 
ELMM.....: 0.1032 
GLMM.....: 0.1036 
DeepGen..: 0.2663 

SAM for Endmembers
PLMM.....: 0.0440 
ELMM.....: 0.0398 
GLMM.....: 0.0355 
DeepGen..: 0.1819 

RMSE for R
FCLS.....: 0.0350 
PLMM.....: 0.0007 
ELMM.....: 0.0321 
GLMM.....: 0.0235 
DeepGen..: 0.2207 

Time FCLS....: 0.73
Time PLMM....: 131.77
Time ELMM....: 60.20
Time GLMM....: 52.35
Time Proposed: 179.39

**Comment by ricardoborsoi** [comment]
Dear Li,
The performance can fluctuate depending on the initialization and training of the autoencoders, so there may be differences with different tensorflow/keras versions, but the performance I obtain on tensorflow2 does not deviate very much from the original one. The original models were trained using tensorflow1 backend with random seed=1, but the initialization may be different for tensorflor 2. I would suggest trying to check if there is some problem with keras when training the autoencoders (python/vae_keras_ems_train.py), fixing an initialization for the tensorflow backend of keras (monitoring the losses for the autoencoders, should be about 150/140/110), and checking if the issue also occur in other  examples. A good metric to monitor is the image reconstruction error ("RMSE for R"), which should not be much higher than that of FCLS.

**Comment by HE-SHUANG-LI** [comment]
Dear Borsoi,
  Thanks for your timely help, I successfully solve the problem following the possible methods what you said. It is 
the model initailization problem caused by Tensorflow and Keras version mismatch. After the repeated environment matching tests, the loss  arrive about 150/140/110 and finally the result has a good performence.
  In the end, thank you again for your reliable help.

**Comment by ricardoborsoi** [comment]
Dear Li,
I'm glad that you have been able to solve the problem and that the suggestions have been useful!

**Comment by khalidcoding** [comment]
good evening, sorry for intruding, I get this error everytime I run this code and it says this 
<img width="1183" alt="Screenshot 2023-02-02 at 03 52 40" src="https://user-images.githubusercontent.com/123639564/216204322-8fec15b6-13b3-4308-b9ed-01307a149890.png">
how can I fix it knowing that I'm using macOs (matlab_R2022b)

**Comment by ricardoborsoi** [comment]
Dear @khalidcoding , thanks for reporting this issue. It is caused by a problem when Matlab tries using some pre-compiled MEX functions from the MinFunc library in Mac OSX, which is used during the optimization. 

I implemented a fix by disabling the use of MEX functions on MAC, could you please download the updated version of the repository and let me know if it works? Thanks.

**Comment by khalidcoding** [comment]
dear @ricardoborsoi, it's working thank you so much.
one more question if you could help me, in example1.m in show results rmse_A_FCLS there is a variable named alpha_cube which is not loaded or something would you explain it for me ? 
thanks

**Comment by ricardoborsoi** [comment]
@khalidcoding, I'm glad it works now!
The variable named "alphas_cube" is the ground truth for the abundances which is used to compute the error metrics and plot the results, it is loaded from the file "data/data_ex1.mat" in the beginning of the code.
